pkg/virtctl/vmexport/vmexport.go

```diff
@@ -0,0 +1,639 @@
+		if err := handleDeleteFlags(args); err != nil {
+			return err
+		}
+	case DOWNLOAD:
+		exportFunction = downloadVirtualMachineExport
+		if err := handleDownloadFlags(args); err != nil {
+			return err
+		}
+	default:
+		return fmt.Errorf("Invalid function '%s'", funcName)
+	}
+
+	// VirtualMachineExport name
+	vmexportName = args[1]
+
+	return nil
+}
+
+// getVirtualMachineExport serves as a wrapper to get the VirtualMachineExport object
+func getVirtualMachineExport(client kubecli.KubevirtClient, namespace string) (*exportv1.VirtualMachineExport, error) {
+	vmexport, err := client.VirtualMachineExport(namespace).Get(context.TODO(), vmexportName, metav1.GetOptions{})
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			return nil, nil
+		}
+		return nil, err
+	}
+
+	return vmexport, nil
+}
+
+// createVirtualMachineExport serves as a wrapper to create the virtualMachineExport object and, if needed, do error handling
+func createVirtualMachineExport(client kubecli.KubevirtClient, namespace string) error {
+	vmexport, err := getVirtualMachineExport(client, namespace)
+	if err != nil {
+		return err
+	}
+	if vmexport != nil {
+		return fmt.Errorf("VirtualMachineExport '%s/%s' already exists", namespace, vmexportName)
+	}
+
+	// Generate/get secret to be used with the vmexport
+	secret, err := getOrCreateTokenSecret(client, namespace)
+	if err != nil {
+		return err
+	}
+
+	vmexport = &exportv1.VirtualMachineExport{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      vmexportName,
+			Namespace: namespace,
+		},
+		Spec: exportv1.VirtualMachineExportSpec{
+			TokenSecretRef: secret.Name,
+			Source: k8sv1.TypedLocalObjectReference{
+				APIGroup: &k8sv1.SchemeGroupVersion.Group,
+				Kind:     resourceKind,
+				Name:     resourceName,
+			},
+		},
+	}
+
+	_, err = client.VirtualMachineExport(namespace).Create(context.TODO(), vmexport, metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("VirtualMachineExport '%s/%s' created succesfully\n", namespace, vmexportName)
+	return nil
+}
+
+// deleteVirtualMachineExport serves as a wrapper to delete the virtualMachineExport object
+func deleteVirtualMachineExport(client kubecli.KubevirtClient, namespace string) error {
+	if err := client.VirtualMachineExport(namespace).Delete(context.TODO(), vmexportName, metav1.DeleteOptions{}); err != nil {
+		if !k8serrors.IsNotFound(err) {
+			return err
+		}
+		return fmt.Errorf("VirtualMachineExport '%s/%s' does not exist", namespace, vmexportName)
+	}
+
+	fmt.Printf("VirtualMachineExport '%s/%s' deleted succesfully\n", namespace, vmexportName)
+
+	// If it exists, attempt to delete the related secret
+	if err := deleteExportSecret(client, namespace); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// downloadVirtualMachineExport handles the process of downloading the requested volume from a VirtualMachineExport object
+func downloadVirtualMachineExport(client kubecli.KubevirtClient, namespace string) error {
+	if shouldCreate {
```

Comment:
That's a good idea but there are several flags that we want to validate before creating the export (snapshop|pvc|vm). The create flag is allowing us to do this validation before starting the download process, so I still think it's kinda useful.

Reply:
I am saying that we can infer the create flag from the existence of the VMExport. If it doesn't exist, create is true, if it does exist, create is false.

Reply:
So I've been testing some options involving deleting the `--create` flag but I always end up with uglier code or needing some additional logic that I'm not too happy with. Flag validation is the first thing we do, so this would require first getting the client, checking if the vmexport exists, and then doing the validation. Another option would be just to assume `--create` to be true if --pvc, --vm or --snapshot exists and skip additional validation, however, I guess that could confuse some users.

I know it's probably not a big deal, but after attempting to commit this change several times, I always find the `--create` flag more pretty and (maybe) easier to understand than other options.